modules/sdk-coin-sol/src/sol.ts

```diff
@@ -559,6 +562,9 @@ export class Sol extends BaseCoin {
    */
   async recover(params: RecoveryOptions): Promise<SolTx> {
     const isUnsignedSweep = !params.userKey && !params.backupKey && !params.walletPassphrase;
+    const startingDerivationIndex = !_.isUndefined(params.startingScanningIndex) ? params.startingScanningIndex : 0;
+    const endingDerivationIndex = !_.isUndefined(params.scan) ? params.scan : 4294967295; // largest positive 32 bit int
```

Comment:
this should be `params.scan + startingDerivationIndex`.

`scan` defines the number of addresses we want to scan through before halting.

also, lets default scanning to 20 or something.